types/lodash/common/common.d.ts

```diff
@@ -245,7 +245,10 @@ declare module "../index" {
     type PropertyName = string | number | symbol;
     type PropertyPath = Many<PropertyName>;
     /** Common interface between Arrays and jQuery objects */
-    type List<T> = ArrayLike<T>;
+    interface List<T> { // Don't use ArrayLike as it's readonly
```

Comment:
This is a major change that will affect lots of things. Instead can you create a new `MutableList` type which is only used in the mutating functions?